.github/pull_request_template.md

```diff
@@ -1,110 +1,98 @@
+Current branch: BRANCH_NAME
 
 ## Related Issue
-<!-- IMPORTANT: Please verify this issue number is correct and exists -->
-<!-- Use the format: Closes #123 or Fixes #123 to automatically close the issue when this PR is merged -->
-Closes #
+<!-- IMPORTANT: Please verify this issue number is correct and exists to automatically close the issue when this PR is merged -->
+Closes #ISSUE_NUMBER
 
 ## Feature Description
 <!-- Provide a clear and concise description of the feature being implemented -->
-<!-- Example: This PR adds user authentication via OAuth2 with GitHub provider, allowing users to log in with their GitHub accounts. This includes the login flow, user profile creation, and session management. -->
 
 ## Implementation Details
 <!-- Provide a technical overview of how the feature was implemented -->
 <!-- Include architectural decisions and design patterns used -->
 <!-- Example:
-- Implemented using the Strategy pattern to allow for multiple auth providers in the future
-- Used JWT for session management with a 24-hour expiration
-- Created a middleware layer to handle authentication checks
+- Used Strategy pattern for multiple auth providers
+- Middleware for authentication checks
+-->
+
+**Important Notes (please address or acknowledge):**
+<!--
+- Mention any CI/CD failing checks unrelated to this PR
+- List any TODO items that are still pending and not closed by this PR
 -->
 
 ## Changes Made
-<!-- List the significant changes made to implement this feature -->
+<!-- List key changes made for this feature -->
 <!-- Example:
-- Added AuthService class to handle authentication flow
-- Created new database migrations for user table
-- Updated configuration to support OAuth settings
-- Added middleware for authenticated routes
-- Created login/logout endpoints
+- Added AuthService class to handle authentication
+- Created user table migrations
 -->
--
--
--
 
 ## Testing Performed
-<!-- Describe the testing you've done to validate the feature -->
-<!-- Include both automated and manual testing -->
+<!-- Describe the testing you've done both automated and manual to validate the feature -->
 <!-- Example:
-- Added unit tests for AuthService with 95% coverage
-- Added integration tests for authentication flow
-- Manually tested login flow on Chrome, Firefox, and Safari
-- Verified error handling for invalid credentials
+- Manually tested pre-commit passing on all files
+- Verified newly added workflows pass in CI/CD
+- Tested any new tool/command added
 -->
 
-## Pre-commit and Testing Instructions
-<!-- Follow these steps to run pre-commit hooks and tests before submitting -->
+## Testing Instructions
+<!-- Follow these steps to run and test before submitting -->
 
-**Running Pre-commit Hooks:**
-```bash
-# These are basic defaults for testing.
-# Please adjust as needed for complete testing of your changes
+**Common Testing Instructions**
```

Comment:
I would like to see groups of workflows of commands like:

**Test github action** 

```just 
just some_command
just some_second_command
```


**Test NAME_SUBJECT** 

```just 
just some_command
```

- NAME A MANUAL STEP 1
- NAME MANUAL STEP 2

```just
just some_second_command
```

@husnain067  Please work with @Tatianah3 's proposal on what workflows have testing.